google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -965,6 +969,20 @@ protected final SpannerRpc.ResultStreamConsumer consumer() {
     public void setCall(SpannerRpc.StreamingCall call, boolean withBeginTransaction) {
       this.call = call;
       this.withBeginTransaction = withBeginTransaction;
+      ApiCallContext callContext = call.getCallContext();
+      Duration streamWaitTimeout = callContext == null ? null : callContext.getStreamWaitTimeout();
+      if (streamWaitTimeout != null) {
+        // Determine the timeout unit to use. This reduces the precision to seconds if the timeout
+        // value is more than 1 second, which is lower than the precision that would normally be
+        // used by the stream watchdog (which uses a precision of 10 seconds by default).
```

Comment:
> This reduces the precision to seconds if the timeout value is more than 1 second.

Not sure if I understand the above correctly.

IIUC, if the timeout value is 2 seconds 1000 nanoseconds (which is greater than 1 second), it'll set `streamWaitTimeoutValue` to `2` and `streamWaitTimeoutUnit` to `TimeUnit.SECONDS`. It'll just trim the nanoseconds, if there's any.

Can we update this comment if the above understanding is correct?

Let me know if I misunderstood something.

Reply:
Okay I think I get it now. The comment makes sense now. Resolving.